File: FocusOnGame.Api/frontend/src/services/PathwayService.js
const API_URL = "http://localhost:5265/api/pathways";

export const pathwayService = {

  async getPathways(idUser) {
    const res = await fetch(`${API_URL}/${idUser}`);
    return res.json();
  },

  async createPathway(pathway) {
    const res = await fetch(API_URL, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(pathway)
    });

    return res.json();
  },

  async updatePathway(idPathway, pathway) {
    const res = await fetch(`${API_URL}/${idPathway}`, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(pathway)
    });

    return res.json();
  },

  async deletePathway(idPathway) {
    await fetch(`${API_URL}/${idPathway}`, {
      method: "DELETE"
    });
  }
};

File: FocusOnGame.Api/backend/Services/PathwayService.cs
using Microsoft.EntityFrameworkCore;
using FocusOnGame.Api.Data;
using FocusOnGame.Api.Dtos;
using FocusOnGame.Api.Models;

namespace FocusOnGame.Api.Services
{
    public class PathwayService
    {
        private readonly AppDbContext _context;

        public PathwayService(AppDbContext context)
        {
            _context = context;
        }

        public async Task<List<PathwayDto>> GetPathways(Guid idUser)
        {
            return await _context.Pathways
                .Where(p => p.IdUser == idUser)
                .Select(g => new PathwayDto
                {
                    Id = g.Id,
                    Name = g.Name,
                    IdUser = g.IdUser
                })
                .ToListAsync();
        }

        public async Task<PathwayDto> CreatePathway(PathwayDto dto)
        {
            var pathway = new Pathway
            {
                Name = dto.Name,
                IdUser = dto.IdUser
            };

            _context.Pathways.Add(pathway);

            await _context.SaveChangesAsync();

            dto.Id = pathway.Id;

            return dto;
        }

        public async Task<PathwayDto?> UpdatePathway(Guid id, PathwayDto dto)
        {
            var pathway = await _context.Pathways.FindAsync(id);

            if (pathway == null)
                return null;

            pathway.Name = dto.Name;

            await _context.SaveChangesAsync();

            return dto;
        }

        public async Task<bool> DeletePathway(Guid id)
        {
            var pathway = await _context.Pathways.FindAsync(id);

            if (pathway == null)
                return false;

            _context.Games.RemoveRange(_context.Games.Where(g => g.IdPathway == id)
);
            _context.Pathways.Remove(pathway);

            await _context.SaveChangesAsync();

            return true;
        }
    }
}

